src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -350,9 +370,13 @@ private unsafe T[] GetListByName(string name, Span<byte> utf8Name, MemberListTyp
                     if (name.Length != 0)
                         Encoding.UTF8.GetBytes(name, utf8Name);
 
+                    var (metadataUpdateDeletedAttributeRuntimeType, declaringModule) = MetadataUpdater.IsSupported && RuntimeTypeMetadataUpdateHandler.FilterDeletedMembers
+                        ? (typeof(System.Runtime.CompilerServices.MetadataUpdateDeletedAttribute).UnderlyingSystemType as RuntimeType, ReflectedType.GetRuntimeModule())
+                        : default;
+
                     fixed (byte* pUtf8Name = utf8Name)
                     {
-                        Filter filter = new Filter(pUtf8Name, utf8Name.Length, listType);
+                        Filter filter = new Filter(pUtf8Name, utf8Name.Length, listType, metadataUpdateDeletedAttributeRuntimeType, declaringModule);
```

Comment:
It does not look right to pass `declaringModule` to the filter. Single instance of the filter can receive methods from multiple different modules.

Notice that the second loop in `PopulateMethods` walks inheritance hierarchy.